pallets/rc-migrator/src/accounts.rs

```diff
@@ -519,25 +551,80 @@ impl<T: Config> AccountsMigrator<T> {
 	///
 	/// Should be executed once before the migration starts.
 	pub fn obtain_rc_accounts() -> Weight {
-		for (channel_id, _info) in hrmp::HrmpChannels::<T>::iter() {
+		let mut reserves = sp_std::collections::btree_map::BTreeMap::new();
+		let mut update_reserves = |id, deposit| {
+			if deposit == 0 {
+				return;
+			}
+			reserves.entry(id).and_modify(|e| *e += deposit).or_insert(deposit);
+		};
+
+		for (channel_id, info) in hrmp::HrmpChannels::<T>::iter() {
 			let sender: T::AccountId = channel_id.sender.into_account_truncating();
-			RcAccounts::<T>::insert(sender, AccountStateFor::<T>::Preserve);
+			update_reserves(sender, info.sender_deposit);
 
 			let recipient: T::AccountId = channel_id.recipient.into_account_truncating();
-			RcAccounts::<T>::insert(recipient, AccountStateFor::<T>::Preserve);
+			update_reserves(recipient, info.recipient_deposit);
+		}
+
+		for (channel_id, info) in hrmp::HrmpOpenChannelRequests::<T>::iter() {
+			let sender: T::AccountId = channel_id.sender.into_account_truncating();
+			update_reserves(sender, info.sender_deposit);
 		}
 
 		for (_, info) in Paras::<T>::iter() {
-			RcAccounts::<T>::insert(
-				info.manager,
-				// TODO: we can use `Part` variation to keep only the reserved part on RC
-				// for now for simplicity we preserve the whole account on RC
-				AccountStateFor::<T>::Preserve,
-			);
+			update_reserves(info.manager, info.deposit);
 		}
 
-		// TODO: should we consider `hrmp::HrmpOpenChannelRequests` or we can just clean up it
-		// before the migration.
+		for (id, rc_reserved) in reserves {
+			let account_entry = SystemAccount::<T>::get(&id);
+			let free = <T as Config>::Currency::balance(&id);
+			let total_frozen = account_entry.data.frozen;
+			let total_reserved = <T as Config>::Currency::reserved_balance(&id);
+			let total_hold = pallet_balances::Holds::<T>::get(&id)
+				.into_iter()
+				// we do not expect more holds
+				.take(5)
+				.map(|h| h.amount)
+				.sum::<T::Balance>();
+
+			let rc_ed = <T as Config>::Currency::minimum_balance();
+			let ah_ed = T::AhExistentialDeposit::get();
+
+			// we prioritize the named holds over the unnamed reserve. if any named holds we will
+			// send them to the AH and keep up to the `rc_reserved` on the RC.
+			let rc_reserved = rc_reserved.min(total_reserved.saturating_sub(total_hold));
```

Comment:
This should be a`defensive_min`, or? We dont expect there to be less than `rc_reserved` in reserves on the RC.